blocklylib-core/src/main/java/com/google/blockly/android/ui/TrashCanView.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.blockly.android.ui;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.drawable.Drawable;
+import android.support.v4.content.ContextCompat;
+import android.util.AttributeSet;
+import android.view.DragEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+import android.widget.ImageView;
+
+import com.google.blockly.android.AbstractBlocklyActivity;
+import com.google.blockly.android.R;
+import com.google.blockly.android.control.BlocklyController;
+
+/**
+ * Drop target view for deleting blocks via a drag gesture, with animation.
+ * <p/>
+ * This view has two layout attributes, {@code closedIcon} and {@code openedIcon}. Each is a
+ * reference to a drawable resource for one of the two trash states, closed (default/idle state) and
+ * opened (pending drop during drag).
+ */
+public class TrashCanView extends FrameLayout {
```

Comment:
A simpler way to do this might be to use a standard ImageView and add a drag listener that sets the highlight or selected state for the view on drag entered.